src/Transition/CreateNewIDFUsingRulesV9_0_0.f90

```diff
@@ -442,7 +455,90 @@ SUBROUTINE CreateNewIDFUsingRules(EndOfFile,DiffOnly,InLfn,AskForInput,InputFile
               ! If your original object starts with P, insert the rules here
 
               ! If your original object starts with R, insert the rules here
-
+              CASE('RUNPERIOD')
+                CALL GetNewObjectDefInIDD(ObjectName,NwNumArgs,NwAorN,NwReqFld,NwObjMinFlds,NwFldNames,NwFldDefaults,NwFldUnits)
+                ! spend some time mining out the state of the run period object
+                RunPeriodStartYear%wasSet = .FALSE.
+                IF (CurArgs >= 14) THEN
+                  IF (TRIM(InArgs(14)) /= Blank) THEN
+                    RunPeriodStartYear%wasSet = .TRUE.
+                    RunPeriodStartYear%originalValue = InArgs(14)
+                  END IF
+                END IF
+                RunPeriodRepeated%wasSet = .FALSE.
+                IF (CurArgs >= 12) THEN
+                  IF (TRIM(InArgs(12)) /= Blank) THEN
+                    RunPeriodRepeated%wasSet = .TRUE.
+                    RunPeriodRepeated%originalValue = InArgs(12)
+                  END IF
+                END IF
+                ! Now start writing some object data out
+                ! Name, BeginMonth and BeginDay are the same
+                OutArgs(1:3) = InArgs(1:3) 
+                ! Start year is weird
+                IF (RunPeriodStartYear%wasSet) THEN
+                  OutArgs(4) = RunPeriodStartYear%originalValue
+                ELSE IF (RunPeriodRepeated%wasSet) THEN
+                  READ(OutArgs(2), *) MonthNumber
+                  READ(OutArgs(3), *) DayNumber
+                  IF (TRIM(InArgs(6)) /= Blank) THEN
+                    YearNumber = FindYearForWeekDay(MonthNumber, DayNumber, InArgs(6))
+                  ELSE
+                    YearNumber = FindYearForWeekDay(MonthNumber, DayNumber, "SUNDAY")
+                  END IF
+                  WRITE(OutArgs(4), *) YearNumber
```

Comment:
@Myoldmopar `RoundSigDigits(YearNumber,0)` should solve the extra spaces problem.